app/events/rtjc.go

```diff
@@ -54,17 +58,29 @@ func (l Rtjc) Listen(ctx context.Context) {
 		if !strings.HasPrefix(msg, "⚠") {
 			return
 		}
-		title, txt, err := l.summary(msg)
+
+		sendMessages, err := l.getSummaryMessages(msg)
 		if err != nil {
 			log.Printf("[WARN] can't get summary, %v", err)
 			return
 		}
-		if txt == "" {
-			log.Printf("[WARN] empty summary for %q", msg)
-			return
-		}
-		if serr := l.Submitter.Submit(ctx, title+"\n\n"+txt, false); serr != nil {
-			log.Printf("[WARN] can't send summary, %v", serr)
+
+		// Rate limit to 15 messages per 2 minutes.
+		// Telegram asks 30 sec of waiting after sending 20 messages
+		rl := rate.NewLimiter(rate.Every(time.Minute*2), 15)
+		i := 0
+		for sendMsg := range sendMessages {
```

Comment:
as noted already - there are not "messages" plural here, so no need in neither loop or limiter

Reply:
There may be multiple messages for user's links. This is the reason of this cycle